### PR TITLE
[Paddle]Add chapter_computational-performance

### PR DESCRIPTION
**Pull request by tngt**

### CODE REVIEW

**Comment by w5688414** [comment]
image

I was wondering why async-computation.md will run on cpu? but the code needs to run on gpu, is configuration right?

**Comment by cheungdaven** [comment]
> image
> 
> I was wondering why async-computation.md will run on cpu? but the code needs to run on gpu, is configuration right? the program works fine on my local machine
> 
> @[cheungdaven](https://github.com/cheungdaven)

Could you try to put d2l.try_gpu() in th beginning of this notebook? @w5688414

**Review by cheungdaven — CHANGES_REQUESTED** [review]

File: chapter_computational-performance/async-computation.md [hunk 1]
```diff
@@ -94,6 +136,10 @@ with d2l.Benchmark():
 广义上说，PyTorch有一个用于与用户直接交互的前端（例如通过Python），还有一个由系统用来执行计算的后端。如 :numref:`fig_frontends`所示，用户可以用各种前端语言编写PyTorch程序，如Python和C++。不管使用的前端编程语言是什么，PyTorch程序的执行主要发生在C++实现的后端。由前端语言发出的操作被传递到后端执行。后端管理自己的线程，这些线程不断收集和执行排队的任务。请注意，要使其工作，后端必须能够跟踪计算图中各个步骤之间的依赖关系。因此，不可能并行化相互依赖的操作。
 :end_tab:
 
+:begin_tab:`pytorch`
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/async-computation.md:139]: Done

File: chapter_computational-performance/async-computation.md [hunk 2]
```diff
@@ -134,6 +188,15 @@ z
 让我们看看这在实践中是如何运作的。
 :end_tab:
 
+:begin_tab:`mxnet`
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/async-computation.md:191]: Done

File: chapter_computational-performance/async-computation.md [hunk 3]
```diff
@@ -144,12 +207,28 @@ with d2l.Benchmark('wait_to_read'):
     b.wait_to_read()
 ```
 
+```{.python .input}
+#@tab paddle
+with d2l.Benchmark('等待计算完成'):
+    b = paddle.dot(a, a)
+    paddle.device.cuda.synchronize()
+
+with d2l.Benchmark('不等待计算完成'):
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/async-computation.md:216]: Done

File: chapter_computational-performance/async-computation.md [hunk 4]
```diff
@@ -178,14 +272,26 @@ with d2l.Benchmark('asynchronous'):
     npx.waitall()
 ```
 
-:begin_tab:`mxnet`
+```{.python .input}
+#@tab paddle
+with d2l.Benchmark('同步执行'):
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/async-computation.md:277]: Done

File: chapter_computational-performance/async-computation.md [hunk 5]
```diff
@@ -178,14 +272,26 @@ with d2l.Benchmark('asynchronous'):
...
+    for _ in range(10000):
+        y = x + 1
+        paddle.device.cuda.synchronize()
+
+with d2l.Benchmark('异步执行'):
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/async-computation.md:282]: Done

File: chapter_computational-performance/auto-parallelism.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
 # 自动并行
 :label:`sec_auto_para`
 
-深度学习框架（例如，MxNet和PyTorch）会在后端自动构建计算图。利用计算图，系统可以了解所有依赖关系，并且可以选择性地并行执行多个不相互依赖的任务以提高速度。例如， :numref:`sec_async`中的 :numref:`fig_asyncgraph`独立初始化两个变量。因此，系统可以选择并行执行它们。
+深度学习框架（例如，MxNet、飞桨和PyTorch）会在后端自动构建计算图。利用计算图，系统可以了解所有依赖关系，并且可以选择性地并行执行多个不相互依赖的任务以提高速度。例如， :numref:`sec_async`中的 :numref:`fig_asyncgraph`独立初始化两个变量。因此，系统可以选择并行执行它们。
 
-通常情况下单个操作符将使用所有CPU或单个GPU上的所有计算资源。例如，即使在一台机器上有多个CPU处理器，`dot` 操作符也将使用所有CPU上的所有核心（和线程）。这样的行为同样适用于单个GPU。因此，并行化对于单设备计算机来说并不是很有用，而并行化对于多个设备就很重要了。虽然并行化通常应用在多个GPU之间，但增加本地CPU以后还将提高少许性能。例如， :cite:`Hadjis.Zhang.Mitliagkas.ea.2016`则把结合GPU和CPU的训练应用到计算机视觉模型中。借助自动并行化框架的便利性，我们可以依靠几行Python代码实现相同的目标。更广泛地考虑，我们对自动并行计算的讨论主要集中在使用CPU和GPU的并行计算上，以及计算和通信的并行化内容。
+通常情况下单个操作符将使用所有CPU或单个GPU上的所有计算资源。例如，即使在一台机器上有多个CPU处理器，`dot` 操作符也将使用所有CPU上的所有核心（和线程）。这样的行为同样适用于单个GPU。因此，并行化对于单设备计算机来说并不是很有用，而并行化对于多个设备就很重要了。虽然并行化通常应用在多个GPU之间，但增加本地CPU以后还将提高少许性能。例如，`Hadjis.Zhang.Mitliagkas.ea.2016`则把结合GPU和CPU的训练应用到计算机视觉模型中。借助自动并行化框架的便利性，我们可以依靠几行Python代码实现相同的目标。更广泛地考虑，我们对自动并行计算的讨论主要集中在使用CPU和GPU的并行计算上，以及计算和通信的并行化内容。
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/auto-parallelism.md:6]: Done

File: chapter_computational-performance/hybridize.md [hunk 7]
```diff
@@ -148,6 +152,32 @@ net = get_net()
...
+```{.python .input}
+#@tab paddle
+from d2l import paddle as d2l
+import paddle
+from paddle import nn
+from paddle.jit import to_static
+from paddle.static import InputSpec
+
+# 生产网络的工厂模式
+# 因为飞桨动态图太快了，需要增加更多的线形图来对比，比如42层。
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/hybridize.md:164]: Done

File: chapter_computational-performance/multiple-gpus.md [hunk 8]
```diff
@@ -266,6 +337,21 @@ print('load into', devices)
 print('output:', split)
 ```
 
+```{.python .input}
+#@tab paddle
+def paddlescatter(XY, devices): 
+    xy = int(XY.shape[0]/len(devices)) # 根据GPU数目计算分块大小
```
[the remark this answers is not in the document]
Comment (w5688414) [reply chapter_computational-performance/multiple-gpus.md:343]: Done

**Comment by cheungdaven** [comment]
@w5688414 could you fix the conflicts? thanks!

**Comment by w5688414** [comment]
> @w5688414 could you fix the conflicts? thanks!

the conflicts was fixed, it works fine again, can you review this chapter again? @cheungdaven

**Pull request merged**